### DISCUSSION TITLE
Why are my changes to an extension not going through?

### DISCUSSION THREAD

**Discussion by sboudreaux96** [eclipse-theia/theia#9482] [Q&A] [answered]
Hello - I have gone through the extension and plugin tutorials for Theia and have made a front end widget extension via the theia-generator. I have made a small adjustment to a string that is displayed when a button on the widget is clicked and the changes to not show when I re-run Theia. I thought it could be a cache problem but I have cleared my browser cache and re-ran it, which gave the same problem. Below is the exact string that I changed in the following file path: C:\Users\username\theia\widget-extension\widget-extension\src\browser\widget-extension-widget.tsx
![image](https://user-images.githubusercontent.com/83592423/118182334-dddd3f80-b406-11eb-8fdd-ed9efb085d64.png)

**Comment by vince-fugnitto** [comment, marked as answer]
@sboudreaux96 thank you for the discussion.

Can you please describe the steps you did from the point you modified the code? 
You need to rebuild the changes in the extension, as well as the `browser-example` or `electron-example` respectfully.

**Reply by sboudreaux96** [reply]
how do we rebuild the changes in the extension exactly? I have tried using yarn rebuild:browser in the root directory but that didnt work

**Reply by vince-fugnitto** [reply]
> I have tried using yarn rebuild:browser

`yarn rebuild:browser` rebuilds the native dependencies so you can switch from the `browser` and `electron` target.

> how do we rebuild the changes in the extension exactly

You can simply perform a `yarn` and `yarn start:browser` at the root, or:

You can:
- perform `yarn` in your extension
- perform `yarn` in the `browser-app` or `electron-app`
- perform `yarn start` in either the `browser-app` or `electron-app` (given that their native dependencies are rebuilt)

**Reply by sboudreaux96** [reply]
thank you for your answer @vince-fugnitto - that works!
